contracts/queue/src/contract.rs

```diff
@@ -80,11 +80,12 @@ fn enqueue<S: Storage, A: Api, Q: Querier>(
     value: i32,
 ) -> StdResult<HandleResponse> {
     // find the last element in the queue and extract key
-    let last_key = deps
-        .storage
-        .range(None, None, Order::Descending)?
-        .next()
-        .map(|(k, _)| k);
+    let last_item = deps.storage.range(None, None, Order::Descending)?.next();
+    let last_key = match last_item {
```

Comment:
I wonder if there is a simpler way to do this.

I can't think of one right now, but it would be good to think a bit. (Note this is only when we call `next()` directly, not when we use an iterator.